### PR TITLE
Support configurable delimiter in syslog plugin

### PR DESCRIPTION
**Pull request by abenbachir**
Adding configurable delimiter support as fluentd patch
Adding documentation for patch creation

### CODE REVIEW

File: source/ext/patches/fluentd/in_syslog.patch [hunk 1]
```diff
@@ -0,0 +1,20 @@
+--- ../source/ext/fluentd/lib/fluent/plugin/in_syslog.rb	2019-04-15 15:49:46.112195768 -0700
++++ ../source/ext/fluentd/lib/fluent/plugin/in_syslog.rb.new	2019-04-15 16:04:08.530462598 -0700
+@@ -110,6 +110,8 @@
+     # 13 is the default value of rsyslog and syslog-ng
+     desc 'The default PRI value (0 - 191 are available)'
+     config_param :default_priority, :integer, default: 13
++    desc 'The delimiter value "\n"'
++    config_param :delimiter, :string, default: "\n" # syslog family add "\n" to each message
```
Comment (hestolz) [inline source/ext/patches/fluentd/in_syslog.patch]: "syslog family" do you mean "syslog facility"?

File: docs/How-to-create-patches.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+Let suppose we need to patch in_syslog plugin in fluentd project:
+* Create a copy of in_syslog.rb and name it in_syslog.rb.new
+* Add the required changes to in_syslog.rb.new
+* Navigate to /omsagent/build/ folder
+* Run the diff cmd:
+```
+diff -Naur ../source/ext/fluentd/lib/fluent/plugin/in_syslog.rb ../source/ext/fluentd/lib/fluent/plugin/in_syslog.rb.new > ../source/ext/patches/fluentd/in_syslog.patch
+```
+* Add the following line to configure file:
```
Comment (hestolz) [inline docs/How-to-create-patches.md]: could specify that it is "build/configure"

File: source/ext/patches/fluentd/in_syslog.patch (hunk 1 above)
Comment (abenbachir) [reply source/ext/patches/fluentd/in_syslog.patch]: @hestolz syslog family mean: rsyslog, syslog-ng and others

Comment (abenbachir) [reply source/ext/patches/fluentd/in_syslog.patch]: done

**Pull request merged**